Dfe.Data.SearchPrototype.Data/PostcodeLookupService.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Net.Http.Json;
+using System.Text;
+using System.Text.Json;
+using Dfe.Data.SearchPrototype.Data.Models;
+
+namespace Dfe.Data.SearchPrototype.Data;
+
+public class PostcodeLookupService
+{
+    private IHttpClientFactory _httpClientFactory;
+    public PostcodeLookupService(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<IEnumerable<GeoLocation?>?> LookupPostcodes(IEnumerable<string> postcodes)
+    {
+        var client = _httpClientFactory.CreateClient("PostcodeLookupAPI");
+        var jsonPayload = JsonSerializer.Serialize(new { postcodes = postcodes });
+        HttpResponseMessage response = await client.PostAsync("postcodes", new StringContent(jsonPayload, Encoding.UTF8, "application/json"));
```

Comment:
`minor` - Would `using HttpResponseMessae respons` save the need for `try finally`?

Reply:
good shout - done